runtime/ts/storage/firebase-storage.ts

```diff
@@ -343,7 +337,7 @@ class FirebaseVariable extends FirebaseStorageProvider {
     return this.localModified;
   }
 
-  async _persistChangesImpl() {
+  async _persistChangesImpl(): Promise<void> {
```

Comment:
Can you explain further what's going on here? These functions are returning values, so they should be Promise<boolean>. 

Is this the point you were making about the return values not being used? I don't think we should denote that via an incorrect type on the function.

Reply:
So this change is because _persistChangesImpl() doesn't return a value in some cases.  It appears to me that the actual response is unread, so I marked it as Promise<void>

Reply:
We either need to not return the unused value, or include the return type in the function signature (which might mean inventing a return in the cases where nothing is currently returned).

I'm actually surprised TS doesn't complain about this.

Reply:
No, I mean I'm surprised that it doesn't complain about 
```
async foo() : Promise<void> { return 42; }
```

I'd prefer we just remove the return statements so that the declaration matches the code. Doesn't seem like a major change?

Reply:
The only return used is here:

    if (this.version != version) {
      // A new local modification happened while we were writing the previous one.
      return this._persistChangesImpl();
    }

I removed the return, but I'm not sure exactly what this code is doing.

Reply:
ack, reverted.  I think this is finally good-to-go.